ddl/util/dead_table_lock_checker.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/parser/model"
+	"github.com/pingcap/tidb/util/logutil"
+	"go.etcd.io/etcd/clientv3"
+	"go.uber.org/zap"
+)
+
+const (
+	defaultRetryCnt      = 5
+	defaultRetryInterval = time.Millisecond * 200
+	defaultTimeout       = time.Second
+)
+
+// DeadTableLockChecker uses to check dead table locks.
+// If tidb-server panic or killed by others, the table locks hold by the killed tidb-server maybe doesn't released.
+type DeadTableLockChecker struct {
+	etcdCli *clientv3.Client
+}
+
+// NewDeadTableLockChecker creates new DeadLockChecker.
+func NewDeadTableLockChecker(etcdCli *clientv3.Client) DeadTableLockChecker {
+	return DeadTableLockChecker{
+		etcdCli: etcdCli,
+	}
+}
+
+func (d *DeadTableLockChecker) getAliveServers(ctx context.Context) (map[string]struct{}, error) {
+	var err error
+	var resp *clientv3.GetResponse
+	allInfos := make(map[string]struct{})
+	for i := 0; i < defaultRetryCnt; i++ {
+		select {
+		case <-ctx.Done():
+			return nil, ctx.Err()
+		default:
+		}
+		childCtx, cancel := context.WithTimeout(context.Background(), defaultTimeout)
```

Comment:
Maybe we can use `ctx` instead of `context.Background()`, then we can remove line52-55.

Reply:
great, done.

Reply:
Do we need to remove line52-55？

Reply:
PTAL @crazycs520

Reply:
Sorry, if `childCtx` is done, it will continue instead of returning.